torch/_dynamo/decorators.py

```diff
@@ -552,7 +552,7 @@ def mark_unbacked(t, index, strict=False):
 
 
 @forbid_in_graph
-def mark_dynamic(t, index, *, min=None, max=None):
+def mark_dynamic(t, index, *, min=None, max=None, backend_specializations=None):
```

Comment:
1) Can we do this for mark_unbacked as well?
2) We should probably error somehow if the range on the symint ends up not including the backend_specialization